common/src/main/java/com/wynntils/screens/settings/lists/FeatureList.java

```diff
@@ -0,0 +1,362 @@
+import net.minecraft.sounds.SoundEvents;
+import org.apache.logging.log4j.util.TriConsumer;
+
+public class FeatureList extends ContainerObjectSelectionList<FeatureListEntryBase> {
+    private final WynntilsSettingsScreen settingsScreen;
+    private static final int PADDING = 5;
+
+    private float cachedRenderHeight = 0;
+
+    private boolean draggingScrollButton = false;
+
+    public FeatureList(WynntilsSettingsScreen screen) {
+        super(
+                McUtils.mc(),
+                screen.width,
+                screen.height,
+                (int) (screen.getBarHeight() + 50),
+                screen.height / 10 + Texture.OVERLAY_SELECTION_GUI.height() - 15,
+                25);
+
+        this.settingsScreen = screen;
+
+        this.reAddEntriesWithSearchFilter("");
+
+        this.setRenderBackground(false);
+        this.setRenderTopAndBottom(false);
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        this.renderBackground(poseStack);
+
+        int x = this.getRowLeft();
+        int y = this.y0 + 4 - (int) this.getScrollAmount();
+
+        this.renderList(poseStack, x, y, mouseX, mouseY, partialTick);
+
+        renderScrollButton(poseStack);
+    }
+
+    @Override
+    protected void renderList(PoseStack poseStack, int x, int y, int mouseX, int mouseY, float partialTick) {
+        iterateOnRenderedEntries((featureListEntryBase, index, top) -> {
+            featureListEntryBase.render(
+                    poseStack,
+                    index,
+                    top + 1,
+                    this.getRowLeft(),
+                    this.getRowWidth(),
+                    featureListEntryBase.getRenderHeight(),
+                    mouseX,
+                    mouseY,
+                    Objects.equals(this.getHovered(), featureListEntryBase),
+                    partialTick);
+        });
+    }
+
+    @Override
+    protected void renderBackground(PoseStack poseStack) {
+        float width = getRenderWidth();
+        float height = getRenderHeight();
+        RenderUtils.drawScalingTexturedRect(
+                poseStack,
+                Texture.FEATURE_LIST_BACKGROUND.resource(),
+                0,
+                settingsScreen.getBarHeight(),
+                0,
+                width,
+                height,
+                Texture.FEATURE_LIST_BACKGROUND.width(),
+                Texture.FEATURE_LIST_BACKGROUND.height());
+    }
+
+    @Override
+    public boolean isMouseOver(double mouseX, double mouseY) {
+        return mouseX <= getRenderWidth()
+                && mouseY >= this.settingsScreen.getBarHeight()
+                && mouseY <= this.settingsScreen.getBarHeight() + this.height;
+    }
+
+    @Override
+    protected FeatureListEntryBase getEntryAtPosition(double mouseX, double mouseY) {
+        AtomicReference<FeatureEntry> returnValue = new AtomicReference<>(null);
```

Comment:
Why AtomicReference? Is there a race here?

Reply:
Needed for the code below.